pkg/workload/workload.go

```diff
@@ -175,13 +186,103 @@ func Tuples(count int, fn func(int) []interface{}) BatchedTuples {
 		NumTotal:   count,
 	}
 	if fn != nil {
-		t.Batch = func(batchIdx int) [][]interface{} {
-			return [][]interface{}{fn(batchIdx)}
+		t.FillBatch = func(batchIdx int, cb coldata.Batch, _ *bufalloc.ByteAllocator) {
+			row := fn(batchIdx)
+			// It'd be nice if we could share this between calls to save the work and
+			// the allocation.
+			colTypes := make([]types.T, len(row))
+			for i, datum := range row {
+				if datum == nil {
+					// WIP what do we do here
+					colTypes[i] = types.Bytes
```

Comment:
Shouldn't we know the types up front somehow? I think the `Tuples` interface should return its column types as well.

Reply:
That would also solve the allocations issue right?